src/incubator/TextField/__tests__/index.spec.js

```diff
@@ -83,4 +84,33 @@ describe('TextField', () => {
       expect(validationMessageElement).toBe(null);
     });
   });
+
+  describe('defaultValue', () => {
+    const props = {
+      ...defaultProps,
+      defaultValue: 'someDefaultValue',
+      value: 'someValue'
+    };
+
+    it('value should equal defaultValue on first render when value not given', () => {
+      const renderTree = render(<TestCase {...props} value={undefined}/>);
+
+      renderTree.getByDisplayValue('someDefaultValue');
+    });
+
+    it('value should equal value on first render when given', () => {
+      const renderTree = render(<TestCase {...props} defaultValue={undefined}/>);
+
+      renderTree.getByDisplayValue('someValue');
+    });
+
+    it('on web should reset defaultValue when prop changed after first render', () => {
```

Comment:
I suggest adding the same test without setting `Constants.isWeb` to `true` since the same behavior should also apply on mobile